lib/ui/views/main/home_page.dart

```diff
@@ -49,29 +49,26 @@ class _HomePageState extends ConsumerState<HomePage>
       tabCount++;
     }
 
-    tabController = TabController(
-      length: tabCount,
-      vsync: this,
-    );
-
     WidgetsBinding.instance.addPostFrameCallback((_) async {
-      tabController.animateTo(
-        ref.read(SettingsProviders.settings).mainScreenCurrentPage,
-        duration: Duration.zero,
-      );
-    });
-  }
+      ref.read(mainTabControllerProvider.notifier).initState(this);
 
-  @override
-  void dispose() {
-    tabController.dispose();
-    super.dispose();
+      ref.read(mainTabControllerProvider)!.animateTo(
+            ref.read(SettingsProviders.settings).mainScreenCurrentPage,
+            duration: Duration.zero,
+          );
+    });
   }
 
   @override
   Widget build(BuildContext context) {
     final theme = ref.watch(ThemeProviders.selectedTheme);
 
+    final tabController = ref.watch(mainTabControllerProvider);
```

Comment:
is `dispose` method is necessary ?

Reply:
The provider is auto disposable so whenever we are not watching it it will be destroyed by the garbage collector.